src/stcal/resample/resample.py

```diff
@@ -62,12 +63,16 @@ class Resample:
         "var_poisson": np.float32,
         "err": np.float32,
     }
+    # add default of float32 to support generic variances conveniently
+    output_array_types = defaultdict(lambda: np.float32)
```

Comment:
```suggestion
    output_array_types:defaultdict[str, DTypeLike] = defaultdict(lambda: np.float32)
```

These suggestions should appease mypy.